pkg/cable/wireguard/driver.go

```diff
@@ -0,0 +1,416 @@
+package wireguard
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"sync"
+
+	"github.com/submariner-io/submariner/pkg/cable"
+	"github.com/submariner-io/submariner/pkg/log"
+	"github.com/vishvananda/netlink"
+	"k8s.io/klog"
+
+	"github.com/submariner-io/submariner/pkg/types"
+	"golang.zx2c4.com/wireguard/wgctrl"
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+const (
+	// DefaultListenPort specifies UDP port address of WireGuard
+	DefaultListenPort = 5871
+
+	// DefaultDeviceName specifies name of WireGuard network device
+	DefaultDeviceName = "subwg0"
+
+	// PublicKey is name (key) of publicKey entry in back-end map
+	PublicKey = "publicKey"
+
+	// we assume Linux
+	//deviceType = wgtypes.LinuxKernel
```

Comment:
This needed?

Reply:
This was a placeholder for user-space wireguard support. 
I removed for now and will add comment to README